src/liballoc/collections/mod.rs

```diff
@@ -77,6 +78,22 @@ impl From<LayoutErr> for TryReserveError {
     }
```

Comment:
Maybe it's possible to return the LayoutErr as cause of the TryReserveError when passed here, although it would make the struct bigger.